sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -195,6 +210,42 @@ internal static string GetNewSchemaOperationName(Activity activity, string? url,
             return activity.DisplayName;
         }
 
+        internal static (string? RequestUrl, string? OperationName) GetHttpOperationNameAndUrl(string activityDisplayName, OperationType operationType, ref AzMonList httpMappedTags)
```

Comment:
The `TelemetryItem` and `RequestData` classes were modified to address an issue where the URL formation was being parsed twice. To resolve this new method was introduced that reads the URL and operation name once for each span / telemetry. This approach eliminates redundant parsing and improves the efficiency of the code for ActivityKind.Server for http spans.

Reply:
I think this logic also needs to change. We no longer need to rely on url in new schema as we will have the path available. Simple `http.method + path` concat will be faster/cheaper than `Uri.TryCreate()`.

Can be done as a follow up PR.

Reply:
Yep, thought about it and felt better to be followed in a new PR.